gpii/node_modules/settingsHandlers/src/settingsHandlerUtilities.js

```diff
@@ -338,6 +338,16 @@ gpii.settingsHandlers.writeFile = function (content, options) {
 gpii.settingsHandlers.handleFileSolutionEntry = function (solutionEntry, modifier, parser, isWrite) {
     var options = solutionEntry.options;
 
+    var initialContent = options.initialContent;
+    var createFile = options.createFile;
+    //create file, if file does not exist, and set initial content of the file
+    if (createFile) {
+        var exists = fs.existsSync(options.filename);
+        if (!exists) {
+            fs.writeFileSync(options.filename, initialContent, options.encoding || "utf-8");
```

Comment:
This duplicates code already present in the gpii.settingsHandlers.writeFile utility. As well as this, this strategy will end up writing the file contents twice over. Instead, we should push down this logic into the utility itself. Also, this change to the core logic will require an update to the test cases in order to verify its behaviour, in order to be accepted. It'll be sufficient to add a case which exercises this logic to just one variety of file-based settings handler tests - e.g. an extra entry in https://github.com/GPII/universal/blob/master/gpii/node_modules/settingsHandlers/test/JSONSettingsHandlerTests.js#L47